rootfs/etc/nginx/lua/util/same_site.lua

```diff
@@ -0,0 +1,36 @@
+local _M = {}
+
+-- determines whether to apply a SameSite=None attribute
+-- to a cookie, based on the user agent.
+-- returns: boolean
+--
+-- Chrome 80 treating third-party cookies as SameSite=Strict
+-- if SameSite is missing. Certain old browsers don't recognize
+-- SameSite=None and will reject cookies entirely bearing SameSite=None.
+-- This creates a situation where fixing things for
+-- Chrome >= 80 breaks things for old browsers.
+-- This function compares the user agent against known
+-- browsers which will reject SameSite=None cookies.
+-- reference: https://www.chromium.org/updates/same-site/incompatible-clients
+function _M.same_site_none_compatible(user_agent)
```

Comment:
@BrianKopp I lifted this code to implement a fix on one of my apps and ran into a problem. We have some idiotic code that apparently doesn't send a `User-Agent` header. Running the request through this library causes a 502.

I don't know if there are safeguards in place for this but I thought I should warn you.

Reply:
Awesome thank you for pointing this out! I've added handling for that here https://github.com/kubernetes/ingress-nginx/pull/5091